File: Python/128_longest_consecutive_sequence.py
class Solution:
  def longestConsecutive(self, nums: List[int]) -> int:
    # Convert nums to a hashtable to remove duplicate
    set_nums = set(nums)
    longest = 0
    
    for i in set_nums:
      current = 0
        
      # If the current number is the smallest of the consecutive sequence
      # Start counting up
      if i-1 not in set_nums:
          current += 1
            
          while i + current in set_nums:
            current += 1
                
      longest = max(longest, current)
        
    return longest

File: C++/128_longest_consecutive_sequence.cpp
class Solution {
public:
  int longestConsecutive(vector<int>& nums) {
    std::unordered_set<int> setNums;
    int longest = 0;
    int current;
    
    // Convert nums to a hashtable to remove duplicate
    for (int n : nums) {
      setNums.insert(n);
    }
    
    for (int i : setNums) {
      current = 0;
        
      // If the current number is the smallest of the consecutive sequence
      // Start counting up
      if(setNums.find(i-1) == setNums.end()){
        current++; 
        
        while(setNums.find(i+current) != setNums.end()) current++; 
      }
      longest = max(current, longest);  
    }
    return longest;
  }
};